helm/cluster-azure/templates/_azure_cluster.tpl

```diff
@@ -15,7 +15,7 @@ spec:
   location: {{ .Values.providerSpecific.location }}
   networkSpec:
     subnets:
-      - name: control-plane-subnet
+      - name: {{ if ( include "network.subnets.controlPlane.name" $ ) }}{{ include "network.subnets.controlPlane.name" $ }}{{ else }}control-plane-subnet{{ end }}
```

Comment:
Subnet names get referenced in other places as well, so it would be good to have a single template that we can include, and to do all the if/else calculations there, so we can just use e.g. `{{ include "network.subnets.controlPlane.name" $ ) }}` wherever we need a subnet name.

Reply:
I see subnet name is used in KCP template. It will be also used for private links.

There would be less space for error with a single plain `include`.

Reply:
I can do it , but what do you think of merging this as it is and doing the change in a separate PR right after so we can compare the diffs ? 

the diff in this PR are already so big is hard to spot errors